DataMigration/src/main/java/org/sagebionetoworks/dian/datamigration/SynapseUtil.java

```diff
@@ -0,0 +1,970 @@
+        File exrParticipantsExtractedFolder = createFolderIfNecessary(
+                EXR_PARTICIPANTS_FOLDER_NAME + EXTRACTED_SUFFIX);
+        if (exrParticipantsFolder != null && exrParticipantsExtractedFolder != null) {
+            if (files.exrParticipants != null) {
+                FileHandleAssociation file = files.exrParticipants;
+                File download = new File(EXR_PARTICIPANTS_FOLDER_NAME +
+                        File.separator + file.getAssociateObjectId() + ZIP);
+                System.out.println("Downloading file " + file.getAssociateObjectId() + ".zip");
+                synapse.downloadFile(file, download);
+                System.out.println("Unzipping file " + file.getAssociateObjectId() + ".zip");
+                UnzipUtil.unzip(download.getAbsolutePath(),
+                        exrParticipantsExtractedFolder.getAbsolutePath());
+            }
+        }
+
+        DianParticipantFileFolders folders = new DianParticipantFileFolders();
+        folders.hasdParticipants = hasdParticipantsExtractedFolder;
+        folders.exrParticipants = exrParticipantsExtractedFolder;
+        return folders;
+    }
+
+    /**
+     * @param testSessionExtractedFolder folder containing a list of folders, all of those
+     *                                  folders contain json files of completed test sessions
+     * @return a map of ARC IDs to their test sessions
+     * @throws IOException if a JSON file cannot be read
+     */
+    public static @NonNull Map<String, CompletedTestList> completedTestMap(
+            @NonNull File testSessionExtractedFolder) throws IOException {
+
+        Map<String, CompletedTestList> completedTestMap = new HashMap<>();
+        Map<String, List<TestSession>> map = new HashMap<>();
+
+        int migrationFileCount = 0;
+        List<String> failedToParseList = new ArrayList<>();
+
+        List<String> filePathList = jsonFilePathList(testSessionExtractedFolder);
+
+        // Iterate through the list of folders of days unzipped
+        for (String filePath : filePathList) {
+            File file = new File(filePath);
+            migrationFileCount++;
+            ObjectMapper mapper = new ObjectMapper();
+            InputStream is = new FileInputStream(file);
+            TestSession sessionObj = mapper.readValue(is, TestSession.class);
+
+            if (sessionObj == null) {
+                failedToParseList.add(filePath);
+            } else {
+                List<TestSession> sessions = map.get(sessionObj.participant_id);
+                if (sessions == null) {
+                    sessions = new ArrayList<>();
+                }
+                sessions.add(sessionObj);
+                map.put(sessionObj.participant_id, sessions);
+            }
+        }
+
+        System.out.println("Test sessions parsing complete\nParsed " + migrationFileCount + " files.");
+
+        if (!failedToParseList.isEmpty()) {
+            System.out.println("Failed to parse file(s) " +
```

Comment:
I think because this deals with user data, any exception should warrant immediate program exit so that no data is corrupted.

Reply:
@brucehoff what type of exception would you suggest in this case?  IllegalStateException?  I'm not well versed on exception types.

Reply:
Arguably the error is an `IllegalArgumentException`, but I would not complain if you used `RuntimeException`, which is the parent class of both `IllegalArgumentException` and `IllegalStateException`.  The important thing is to retain the informative message which is currently in the print statement.  When that message is captured and sent out as a notification the recipient will know how to act because they will be told what files are invalid.

Reply:
Will switch to `IllegalArgumentException`